### PR TITLE
Add cats Parallel

### LINKED ISSUE #1034: Add Cats Parallel
In #1028 we have added `BIOParallel`. This gives us all the necessary components to implement `cats.Parallel`

### PR DESCRIPTION
**Pull request by Adriani277**
closes #1034

### CODE REVIEW

**Comment by Adriani277** [comment]
@neko-kai I will attempt to add syntax to `BIOParallel` in a new PR as it may take me a little more time

**Comment by neko-kai** [comment]
@Adriani277 I've left the last few comments on your PR there - https://github.com/Adriani277/izumi/pull/1#pullrequestreview-405119814 - with that done it'll be completely ready to merge. Thank you so much! 🙏

**Comment by neko-kai** [comment]
@Adriani277 Could you please add a test for Parallel conversion in `CatsConversionTest` ?

File: fundamentals/fundamentals-bio/src/main/scala/izumi/functional/bio/BIOCatsConversions.scala [hunk 1]
```diff
@@ -55,7 +56,10 @@ trait BIOCatsConversions7 extends BIOCatsConversions8 {
 trait BIOCatsConversions8 extends BIOCatsConversions9 {
   @inline implicit final def BIOAsyncToAsync[F[+_, +_]](implicit F: BIOAsync[F]): cats.effect.Async[F[Throwable, ?]] with S9 = new BIOCatsAsync[F](F)
 }
-trait BIOCatsConversions9 {
+trait BIOCatsConversions9 extends BIOCatsConversions10 {
+  @inline implicit final def BIOParallelToParallel[F[+_, +_]](implicit F: BIOParallel[F]): cats.Parallel[F[Throwable, ?]] = new BIOCatsParallel[F](F)
```
Comment (Adriani277) [inline fundamentals/fundamentals-bio/src/main/scala/izumi/functional/bio/BIOCatsConversions.scala:60]: @neko-kai The conversion test would not work unless I added this. 
Is this correct?

Comment (neko-kai) [reply fundamentals/fundamentals-bio/src/main/scala/izumi/functional/bio/BIOCatsConversions.scala:60]: @Adriani277 Yep.

**Review by neko-kai — APPROVED** [review]

**Comment by neko-kai** [comment]
@Adriani277 Thank you! 🙏

**Comment by Adriani-Furtado** [comment]
@neko-kai Thank you so much for the help and support. I will attempt to add syntax to BIOParallel and if I can get the hang of it, I will also add for any missing types

**Comment by neko-kai** [comment]
@Adriani277 No problem, looking forward to your contributions 👍

**Pull request merged**